app/models/changeset/pull_request.rb

```diff
@@ -40,11 +40,20 @@ def self.changeset_from_webhook(project, params = {})
 
   def self.valid_webhook?(params)
     data = params['pull_request'] || {}
-    return false unless data['state'] == 'open'
+    return false unless data['state'] == 'open' && valid_action?(params['github']['action'])
+
+    if params['github']['action'] == 'edited'
+      previous_desc = params['github']['changes']['body']['from']
+      return false if previous_desc =~ WEBHOOK_FILTER && data['body'] =~ WEBHOOK_FILTER
+    end
 
     !(data['body'] =~ WEBHOOK_FILTER).nil?
   end
 
+  def self.valid_action?(action)
+    ['opened', 'edited', 'synchronized'].include? action
```

Comment:
It might be worthwhile to either put this `include?` in the `valid_webhook?` method or move the `data['state'] == 'open'` check here. Not sure we're getting a lot of benefit out of moving the single `include?` call into its own method